src/scripts/main.js

```diff
@@ -1,11 +1,22 @@
 'use strict';
 
 function waitFor(element, eventName) {
-  // write your code here
+  return new Promise((resolve) => {
```

Comment:
Consider adding error handling to the promise in case the event listener encounters an issue. This would make the function more robust.